File: backend/core/src/main/kotlin/com/globalairline/core/domain/Airport.kt
package com.globalairline.core.domain

/**
 * Represents an airport in the simulation.
 */
data class Airport(
    val id: Int,
    val iata: String,
    val icao: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val country: String
)

File: frontend/src/types/airport.ts
/**
 * Core airport entity from the backend API.
 */
export interface Airport {
  id: number;
  name: string;
  iata: string | null;
  icao: string | null;
  city: string;
  countryCode: string;
  latitude: number;
  longitude: number;
}

/**
 * Paginated airport list response.
 */
export interface AirportListParams {
  offset?: number;
  limit?: number;
  country?: string;
}

/**
 * Airport search parameters.
 */
export interface AirportSearchParams {
  q: string;
  limit?: number;
}
